tests/ui/unnecessary_map_or.fixed

```diff
@@ -9,9 +9,9 @@ extern crate proc_macros;
 
 fn main() {
     // should trigger
-    let _ = Some(5) == Some(5);
-    let _ = Some(5) != Some(5);
-    let _ = Some(5) == Some(5);
+    let _ = (Some(5) == Some(5));
```

Comment:
These `(` should be removed, shouldn't they?

Reply:
I think this is just an effect of `maybe_par` deciding they should be present. While I'd usually agree, I'm not sure there is a better way to tackle that?

Reply:
`maybe_par` is rather overeager, but `unused_parens` will usually catch them anyway. New lints triggering on fixed code in a cascade is fine